patches/tModLoader/Terraria.ModLoader/ItemLoader.cs

```diff
@@ -968,7 +1000,8 @@ public static void RightClick(Item item, Player player)
 			foreach (var g in HookRightClick.arr)
 				g.Instance(item).RightClick(item, player);
 
-			item.stack--;
+			if (ConsumeItem(item, player))
+				item.stack--;
 			if (item.stack == 0)
```

Comment:
Can be made nested to avoid an unnecessary check from happening if `ConsumeItem` returns false